library/pk_wrap.c

```diff
@@ -220,6 +220,7 @@ static int rsa_verify_wrap(mbedtls_pk_context *pk, mbedtls_md_type_t md_alg,
 
     /* mbedtls_pk_write_pubkey_der() expects a full PK context;
      * re-construct one to make it happy */
+    mbedtls_pk_init(&key);
```

Comment:
We're constructing an `mbedtls_pk_context` containing an ordinary RSA key (`pk_info = mbedtls_rsa_info`). For this case, the new fields (fields other than `pk_info` and `pk_ctx`) are never used. So the code was functionally correct.

However, the code was fragile: if we modify support for RSA, or even if we tweak the implementation of some function used by `mbedtls_pk_write_pubkey_der`, there was a risk that we'd start reading one of the uninitialized fields. So it's good to be more robust here and call `mbedtls_pk_init`. Generally speaking, we shouldn't use a structure without having called its `init` function.

The comment above is misleading nowadays: `mbedtls_pk_write_pubkey_der` does not expect a _full_ PK context in this case, it just needs a working PK context, which for non-opaque RSA keys means just the fields that we set here. So please amend the comment to at the very least remove “full”.

Same goes for the other three.